tests/Data/URITest.php

```diff
@@ -0,0 +1,383 @@
+		$this->assertEquals($uri->authority(),'github.com');
+		$this->assertEquals($uri->host(),'github.com');
+		$this->assertNull($uri->port());
+		$this->assertNull($uri->path());
+		$this->assertEquals($uri->query(),'query_p$,;:A!\'*+()ar_1=val_1&quer?y_par_2=val_2');
+		$this->assertNull($uri->fragment());
+
+		$uri = new ILIAS\Data\URI(self::URI_AUTHORITY_AND_QUERY_2);
+		$this->assertEquals($uri->schema(),'git');
+		$this->assertEquals($uri->authority(),'github.com');
+		$this->assertEquals($uri->host(),'github.com');
+		$this->assertNull($uri->port());
+		$this->assertNull($uri->path());
+		$this->assertEquals($uri->query(),'qu/ery_p$,;:A!\'*+()ar_1=val_1&quer?y_par_2=val_2');
+		$this->assertNull($uri->fragment());
+	}
+
+	/**
+	 * @depends test_init
+	 */
+	public function test_authority_and_fragment()
+	{
+		$uri = new ILIAS\Data\URI(self::URI_AUTHORITY_AND_FRAGMENT);
+		$this->assertEquals($uri->schema(),'git');
+		$this->assertEquals($uri->authority(),'github.com:8080');
+		$this->assertEquals($uri->host(),'github.com');
+		$this->assertEquals($uri->port(),'8080');
+		$this->assertNull($uri->path());
+		$this->assertNull($uri->query());
+		$this->assertEquals($uri->fragment(),'fragment$,;:A!\'*+()ar_1=val_1&');
+	}
+	/**
+	 * @depends test_init
+	 */
+	public function test_authority_path_fragment()
+	{
+		$uri = new ILIAS\Data\URI(self::URI_AUTHORITY_PATH_FRAGMENT);
+		$this->assertEquals($uri->schema(),'git');
+		$this->assertEquals($uri->authority(),'git$,;hub.com:8080');
+		$this->assertEquals($uri->host(),'git$,;hub.com');
+		$this->assertEquals($uri->port(),'8080');
+		$this->assertEquals($uri->path(),'someacc$,;ount/somerepo/somerepo.git');
+		$this->assertNull($uri->query());
+		$this->assertEquals($uri->fragment(),'frag:A!\'*+()arment');
+	}
+
+	/**
+	 * @depends test_init
+	 */
+	public function test_path()
+	{
+		$uri = new ILIAS\Data\URI(self::URI_PATH);
+		$this->assertEquals($uri->schema(),'git');
+		$this->assertEquals($uri->authority(),'git$,;hub.com:8080');
+		$this->assertEquals($uri->host(),'git$,;hub.com');
+		$this->assertEquals($uri->port(),'8080');
+		$this->assertEquals($uri->path(),'someacc$,;ount/somerepo/somerepo.git');
+		$this->assertNull($uri->query());
+		$this->assertNull($uri->fragment());
+		$this->assertEquals($uri->baseURI(),'git://git$,;hub.com:8080/someacc$,;ount/somerepo/somerepo.git');
+	}
+
+	/**
+	 * @depends test_init
+	 */
+	public function test_authority_only()
+	{
+		$uri = new ILIAS\Data\URI(self::URI_AUTHORITY_ONLY);
+		$this->assertEquals($uri->schema(),'git');
+		$this->assertEquals($uri->authority(),'git$,;hub.com');
+		$this->assertEquals($uri->host(),'git$,;hub.com');
+		$this->assertNull($uri->port());
+		$this->assertNull($uri->path());
+		$this->assertNull($uri->query());
+		$this->assertNull($uri->fragment());
+		$this->assertEquals($uri->baseURI(),'git://git$,;hub.com');
+	}
+
+	/**
+	 * @depends test_init
+	 */
+	public function test_no_schema()
+	{
+		try {
```

Comment:
Instead of a `try-catch`-block you could use `@expectedException InvalidArgumentException` in the doc block.()
Or `$this->expectException(InvalidArgumentException::class)`. Whatever is considered best in this project.

See also: https://phpunit.de/manual/6.5/en/writing-tests-for-phpunit.html#writing-tests-for-phpunit.exceptions

This would reduce the code here.

See also the methods below.

Reply:
Please move to `expectException`, our [unit testing guide suggests that](https://github.com/ILIAS-eLearning/ILIAS/tree/trunk/tests#example-solution-1).

Reply:
Ok, that one is covered.